clippy_lints/src/loops/manual_memcpy.rs

```diff
@@ -59,6 +59,7 @@ pub(super) fn check<'tcx>(
                         let rhs = fetch_cloned_expr(rhs);
                         if let ExprKind::Index(base_left, idx_left, _) = lhs.kind
                             && let ExprKind::Index(base_right, idx_right, _) = rhs.kind
+                            && is_index_suggestable(base_left) && is_index_suggestable(base_right)
```

Comment:
I believe this would miss slight variations like `v[i + 1][i]`, I think a good approach would be to check if `base_left/right` contains a reference to `i`

There's [`local_used_in`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/usage/fn.local_used_in.html) though looking at `starts` you may need your own `for_each_expr` that checks for one of several IDs

Reply:
Thank you for your review and suggestions!

I totally agree that my implementation misses some cases, and I'm open to making the necessary changes.

However I realized some cases where containing a reference to `i` might be acceptable like `v[3][3][i]`.
In this case, if my understanding is correct, it could be challenging to differentiate them from others.
I'd appreciate your thoughts on this aspect!

Reply:
If `v[3][3][i]` is on the right hand side then `base_right` should refer to `v[3][3]` assuming I read the code correctly, that would pass the check since it doesn't contain `i`

Reply:
I have made adjustments to the code to check that `base_left/right` does not contain a reference to `i`. Additionally, it seems that `local_used_in` may suffice without implementing a custom `for_each_expr`. Could you please review these changes when you have time?